### ISSUE TITLE
[Feature] D1 Support

### ISSUE THREAD

**Issue by vidiabtc** [cloudflare/workers-rs#221]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

It would be great if workers-rs had the functionality of wrangler@d1. Although I understand that it may not be the best time to do this as D1 is still in private beta, but it would be cool if the D1 integration was ready to go by the time D1 is public!

**Comment by Yrlish** [comment]
I would also be interested in this. I'm also aware that the current state of D1 is in private beta so it may not be doable at the moment.

**Comment by Yrlish** [comment]
> In the future I think we'll have to figure out how we're going to go about private-beta bindings, possible through an `unstable` feature flag?

Yeah this sound like a good idea, some sort of unstable branch for non-GA features.

I was also thinking of an alternative solution to what I was thinking about above. Having two separate workers, one in Rust and one "normal" in Typescript and connecting them via a service binding.

**Comment by yzn-h** [comment]
it is now open alpha ╰(*°▽°*)╯

**Comment by KianNH** [comment]
https://github.com/cloudflare/workers-rs/pull/246

**Comment by Keavon** [comment]
#270

**Comment by frontmesh** [comment]
Will the d1 support be available in 0.0.18 release ? I see that it has been merged, but not yet released.

**Comment by spicyzboss** [comment]
I can't enable d1 feature but I have seen [this commit](https://github.com/cloudflare/workers-rs/commit/e7b41bf5e37d38aa82c7b8fde7fbbaf316f1192c) that not released in 0.0.17. Any updates?
